tests/core/test_mock_signal_backend.py

```diff
@@ -21,12 +21,11 @@
 from ophyd_async.epics.signal.signal import epics_signal_r, epics_signal_rw
 
 
-@pytest.mark.parametrize("connect_mock_mode", [True, False])
```

Comment:
https://github.com/bluesky/ophyd-async/pull/368/files#r1644172513

Please respond to exising review comments before re-requesting a review: either implementing them or leaving a written disagreement. If not it takes more reviews to get to merge-able code.

Reply:
there was enough lack of clarity on other points that I decided to focus on the other parts. the truth is that I still don't understand how is the `connect()` logic supposed to go :( 

maybe I should have left a note about the paremeters point

Reply:
> the truth is that I still don't understand how is the connect() logic supposed to go :(

That's totally fine! I'm happy to explain, in person, or over github.